### ISSUE TITLE
Wallet funds & portfolio apparently missing after upgrade to 0.5.2

### ISSUE THREAD

**Issue by nothingmuch** [bisq-network/bisq#872]
After upgrading to 0.5.2 (using dpkg -i, with a debian stretch Qubes TemplateVM, [upgraded from default debian 8](https://www.qubes-os.org/doc/template/debian/upgrade-8-to-9/)) and starting Bisq, my offer disappeared from the portfolio, and wallet funds were at 0. Downgrading to 0.5.1 fully restored everything to the previous state. Did not investigate further.

**Comment by nothingmuch** [comment]
Here are transcript of my terminal sessions. Note the `^C` towards the end of the first is literal.

I've redacted bitvomit I couldn't confirm was not related to my wallet (I lost patience with code diving on github with regards to the small int after the @, still not sure what that is, and from what I understand the checkpoint is a signature made by my key, didn't confirm if it was only a hash of that or not). Sorry for taking long on this, but this is my first look at the code, so it took a while to grasp the log output.

Perhaps it's noteworthy that the tor hidden service is different in the 0.5.2 and 0.5.1 (though not in my redaction).

## Version 0.5.2 transcript

```
user@bisq:~$ sudo dpkg -i Downloads/Bisq-64bit-0.5.2.deb 
(Reading database ... 97661 files and directories currently installed.)
Preparing to unpack Downloads/Bisq-64bit-0.5.2.deb ...
Removing shortcut
Unpacking bisq (0.5.2) over (0.5.1) ...
Setting up bisq (0.5.2) ...
Adding shortcut to the menu
user@bisq:~$ /opt/Bisq/Bisq 
Successfully removed cryptography restrictions
Jul-17 01:11:28.365 [JavaFX Application Thread] INFO  i.b.g.a.BisqApp: Log files under: /home/user/.local/share/Bisq/bisq 
Jul-17 01:11:28.368 [JavaFX Application Thread] INFO  i.b.c.u.Utilities: System info: os.name=Linux; os.version=4.4.67-13.pvops.qubes.x86_64; os.arch=amd64; sun.arch.data.model=64; JRE=1.8.0_131-b11 (Oracle Corporation); JVM=25.131-b11 (Java HotSpot(TM) 64-Bit Server VM) 
Jul-17 01:11:31.634 [JavaFX Application Thread] INFO  i.b.c.p.HttpClientProvider: FeeProvider with baseUrl http://44mgyoe2b6oqiytt.onion/ 
Jul-17 01:11:31.659 [JavaFX Application Thread] INFO  i.b.c.p.HttpClientProvider: PriceProvider with baseUrl http://44mgyoe2b6oqiytt.onion/ 
Jul-17 01:11:31.738 [JavaFX Application Thread] INFO  i.b.c.a.Version: Version{VERSION=0.5.2, P2P_NETWORK_VERSION=1, LOCAL_DB_VERSION=1, TRADE_PROTOCOL_VERSION=1, BASE_CURRENCY_NETWORK=0, getP2PNetworkId()=10} 
Jul-17 01:11:47.704 [checkIfLocalHostNodeIsRunningThread] INFO  i.b.g.m.MainViewModel: Localhost peer not detected. 
Jul-17 01:11:47.713 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: startBasicServices 
Jul-17 01:11:47.716 [loadEntryMapThread] INFO  i.b.n.p.s.P2PDataStorage: We copy resource to file: resourceFileName=EntryMap_BTC, destinationFile=/home/user/.local/share/Bisq/btc_mainnet/db/EntryMap 
Jul-17 01:11:47.716 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: initP2PNetwork 
Jul-17 01:11:47.776 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: 
checkCryptoSetupResult=false
loadEntryMapResult=false
walletInitialized=false
p2pNetWorkReady=false 
INFO: uname -m call results in:x86_64
Jul-17 01:11:48.105 [loadEntryMapThread] INFO  i.b.n.p.s.PersistedEntryMap: PersistedEntryMap.fromProto size: 4382 
Jul-17 01:11:50.607 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: 
checkCryptoSetupResult=true
loadEntryMapResult=false
walletInitialized=false
p2pNetWorkReady=false 
Jul-17 01:11:56.994 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onTorNodeReady 
Jul-17 01:11:57.002 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: initWalletService 
Jul-17 01:11:57.084 [ STARTING] INFO  i.b.c.b.w.WalletConfig: Wallet directory: /home/user/.local/share/Bisq/btc_mainnet/wallet 
Jul-17 01:11:57.742 [ STARTING] INFO  o.b.c.MnemonicCode: PBKDF2 took 550.3 ms 
Jul-17 01:11:57.832 [ STARTING] INFO  o.b.w.KeyChainGroup: Creating and activating a new HD chain: io.bisq.core.btc.wallet.BtcDeterministicKeyChain@xxx 
Jul-17 01:11:57.852 [ STARTING] INFO  o.b.w.DeterministicKeyChain: 1 keys needed for M/44H/0H/0H/0 = 1 issued + 0 lookahead size + 0 lookahead threshold - 0 num children 
Jul-17 01:11:57.854 [ STARTING] INFO  o.b.w.DeterministicKeyChain: Took 899.9 μs 
Jul-17 01:11:58.084 [ STARTING] INFO  o.b.w.DeterministicKeyChain: 133 keys needed for M/44H/142H/0H/0 = 0 issued + 100 lookahead size + 33 lookahead threshold - 0 num children 
Jul-17 01:11:58.425 [ STARTING] INFO  o.b.w.DeterministicKeyChain: Took 339.3 ms 
Jul-17 01:11:58.426 [ STARTING] INFO  o.b.w.DeterministicKeyChain: 133 keys needed for M/44H/142H/0H/1 = 0 issued + 100 lookahead size + 33 lookahead threshold - 0 num children 
Jul-17 01:11:58.577 [ STARTING] INFO  o.b.w.DeterministicKeyChain: Took 150.0 ms 
Jul-17 01:11:58.718 [ STARTING] INFO  o.b.s.SPVBlockStore: Creating new SPV block chain file /home/user/.local/share/Bisq/btc_mainnet/wallet/bisq.spvchain 
Jul-17 01:11:58.732 [ STARTING] INFO  o.b.c.CheckpointManager: Attempting to initialize a new block store with a checkpoint for time 1499645517 (2017-07-10T00:11:57Z) 
Jul-17 01:11:58.768 [ STARTING] INFO  o.b.c.CheckpointManager: Read 223 checkpoints, hash is xxx 
Jul-17 01:11:58.836 [PeerGroup Thread] INFO  o.b.w.DeterministicKeyChain: 133 keys needed for M/44H/0H/0H/0 = 1 issued + 100 lookahead size + 33 lookahead threshold - 1 num children 
Jul-17 01:11:58.925 [PeerGroup Thread] INFO  o.b.w.DeterministicKeyChain: Took 88.16 ms 
Jul-17 01:11:58.932 [PeerGroup Thread] INFO  o.b.w.DeterministicKeyChain: 133 keys needed for M/44H/0H/0H/1 = 0 issued + 100 lookahead size + 33 lookahead threshold - 0 num children 
Jul-17 01:11:58.989 [PeerGroup Thread] INFO  o.b.w.DeterministicKeyChain: Took 55.79 ms 
Jul-17 01:11:59.105 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: walletsSetup.onInitialized 
Jul-17 01:11:59.106 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: 
checkCryptoSetupResult=true
loadEntryMapResult=false
walletInitialized=true
p2pNetWorkReady=false 
Jul-17 01:12:01.481 [loadEntryMapThread] INFO  i.b.n.p.s.P2PDataStorage: persistedEntryMap size=4382 
Jul-17 01:12:01.527 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: 
checkCryptoSetupResult=true
loadEntryMapResult=true
walletInitialized=true
p2pNetWorkReady=false 
Jul-17 01:12:06.793 [BlockingClient network thread for seed.bitcoin.jonasschnelli.ch/35.184.196.17:8333] WARN  o.b.c.PeerSocketHandler: [35.184.196.17]:8333 -  org.bitcoinj.core.ProtocolException: Received AlertMessage before version handshake is complete.
	at org.bitcoinj.core.Peer.processMessage(Peer.java:478)
	at org.bitcoinj.core.PeerSocketHandler.receiveBytes(PeerSocketHandler.java:182)
	at org.bitcoinj.net.BlockingClient.runReadLoop(BlockingClient.java:122)
	at org.bitcoinj.net.BlockingClient$1.run(BlockingClient.java:81)

org.bitcoinj.core.ProtocolException: Received AlertMessage before version handshake is complete.
	at org.bitcoinj.core.Peer.processMessage(Peer.java:478)
	at org.bitcoinj.core.PeerSocketHandler.receiveBytes(PeerSocketHandler.java:182)
	at org.bitcoinj.net.BlockingClient.runReadLoop(BlockingClient.java:122)
	at org.bitcoinj.net.BlockingClient$1.run(BlockingClient.java:81)
Jul-17 01:12:10.238 [BlockingClient network thread for seed.bitnodes.io/161.67.132.13:8333] INFO  o.b.p.AbstractBitcoinNetParams: Difficulty transition traversal took 68.15 ms 
Jul-17 01:12:29.280 [BlockingClient network thread for seed.bitnodes.io/161.67.132.13:8333] WARN  o.b.c.AbstractBlockChain: Block does not connect: 00000000000000000144fd5b6ebba625238111d44fc2fb876ad1f90df3424fe5 prev 00000000000000000113cae08b605fc061e7c63ba5c8affe8aedb598cb2e012a 
Jul-17 01:12:30.428 [Thread-11] INFO  c.m.t.t.OnionProxyManagerEventHandler: Hidden service xxx.onion:9999 published. 
Jul-17 01:12:30.486 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onHiddenServicePublished 
Jul-17 01:12:30.749 [JavaFX Application Thread] INFO  i.b.n.p.p.g.RequestDataHandler: 
#################################################################
Received 4430 instances of storage payload
TradeStatistics: 4362
MailboxStoragePayload: 6
Alert: 1
Arbitrator: 1
OfferPayload: 60
################################################################# 
Jul-17 01:12:30.940 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onBootstrapComplete 
Jul-17 01:12:30.943 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onRequestingDataCompleted 
Jul-17 01:12:30.943 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: 
checkCryptoSetupResult=true
loadEntryMapResult=true
walletInitialized=true
p2pNetWorkReady=true 
Jul-17 01:12:30.943 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onBasicServicesInitialized 
Jul-17 01:12:34.670 [JavaFX Application Thread] INFO  i.b.n.p.p.g.RequestDataHandler: 
#################################################################
Received 78 instances of storage payload
TradeStatistics: 10
MailboxStoragePayload: 6
Alert: 1
Arbitrator: 1
OfferPayload: 60
################################################################# 
Jul-17 01:12:34.964 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onRequestingDataCompleted 
Jul-17 01:12:34.966 [JavaFX Application Thread] INFO  i.b.c.p.f.FeeService: BTC tx fee: txFeePerByte=80 
^CJul-17 01:13:14.005 [JavaFX Application Thread] ERROR i.b.g.a.BisqApp: Uncaught Exception from thread JavaFX Application Thread 
Jul-17 01:13:14.006 [JavaFX Application Thread] ERROR i.b.g.a.BisqApp: throwableMessage= null 
Jul-17 01:13:14.006 [JavaFX Application Thread] ERROR i.b.g.a.BisqApp: throwableClass= class java.lang.IncompatibleClassChangeError 
Jul-17 01:13:14.013 [JavaFX Application Thread] ERROR i.b.g.a.BisqApp: Stack trace:
java.lang.IncompatibleClassChangeError
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$null$49(GtkApplication.java:139)
	at java.lang.Thread.run(Thread.java:748)
 
java.lang.IncompatibleClassChangeError
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$null$49(GtkApplication.java:139)
	at java.lang.Thread.run(Thread.java:748)
```

## Version 0.5.1 transcript

```
user@bisq:~$ sudo dpkg -i Downloads/Bisq-64bit-0.5.1.deb 
dpkg: warning: downgrading bisq from 0.5.2 to 0.5.1
(Reading database ... 97661 files and directories currently installed.)
Preparing to unpack Downloads/Bisq-64bit-0.5.1.deb ...
Removing shortcut
Unpacking bisq (0.5.1) over (0.5.2) ...
Setting up bisq (0.5.1) ...
Adding shortcut to the menu
user@bisq:~$ /opt/Bisq/Bisq 
Jul-17 01:15:35.402 [JavaFX Application Thread] INFO  i.b.g.a.BisqApp: Log files under: /home/user/.local/share/bisq/bisq 
Jul-17 01:15:35.405 [JavaFX Application Thread] INFO  i.b.c.u.Utilities: System info: os.name=Linux; os.version=4.4.67-13.pvops.qubes.x86_64; os.arch=amd64; sun.arch.data.model=64; JRE=1.8.0_131-b11 (Oracle Corporation); JVM=25.131-b11 (Java HotSpot(TM) 64-Bit Server VM) 
Jul-17 01:15:36.832 [JavaFX Application Thread] INFO  i.b.c.p.HttpClientProvider: FeeProvider with baseUrl http://44mgyoe2b6oqiytt.onion/ 
Jul-17 01:15:36.843 [JavaFX Application Thread] INFO  i.b.c.p.HttpClientProvider: PriceProvider with baseUrl http://44mgyoe2b6oqiytt.onion/ 
Jul-17 01:15:36.867 [JavaFX Application Thread] INFO  i.b.c.s.FileManager: Read from disc: PreferencesPayload 
Jul-17 01:15:37.227 [JavaFX Application Thread] INFO  i.b.c.s.FileManager: Read from disc: UserPayload 
Jul-17 01:15:37.304 [JavaFX Application Thread] INFO  i.b.c.s.FileManager: Read from disc: NavigationPath 
Jul-17 01:15:37.322 [JavaFX Application Thread] INFO  i.b.c.s.FileManager: Read from disc: AddressEntryList 
Jul-17 01:15:37.329 [JavaFX Application Thread] INFO  i.b.c.s.FileManager: Read from disc: TradeStatistics 
Jul-17 01:15:37.527 [Save-file-task-3327] INFO  i.b.c.s.FileManager: Write to disc: PreferencesPayload 
Jul-17 01:15:37.564 [JavaFX Application Thread] INFO  i.b.c.s.FileManager: Read from disc: OpenOffers 
Jul-17 01:15:37.614 [JavaFX Application Thread] INFO  i.b.c.s.FileManager: Read from disc: SequenceNumberMap 
Jul-17 01:15:37.656 [JavaFX Application Thread] INFO  i.b.c.s.FileManager: Read from disc: PeerList 
Jul-17 01:15:37.689 [JavaFX Application Thread] INFO  i.b.c.a.Version: Version{VERSION=0.5.1, P2P_NETWORK_VERSION=1, LOCAL_DB_VERSION=1, TRADE_PROTOCOL_VERSION=1, BASE_CURRENCY_NETWORK=0, getP2PNetworkId()=10} 
Jul-17 01:15:38.128 [Save-file-task-726] INFO  i.b.c.s.FileManager: Write to disc: UserPayload 
Jul-17 01:15:41.238 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: startBasicServices 
Jul-17 01:15:41.248 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: Localhost peer not detected. 
Jul-17 01:15:41.250 [JavaFX Application Thread] INFO  i.b.c.s.FileManager: Read from disc: SequenceNumberMap 
Jul-17 01:15:41.259 [JavaFX Application Thread] INFO  i.b.c.s.FileManager: Read from disc: EntryMap 
Jul-17 01:15:41.551 [JavaFX Application Thread] INFO  i.b.n.p.s.P2PDataStorage: persistedEntryMap size=4362 
Jul-17 01:15:41.562 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: initP2PNetwork 
INFO: uname -m call results in:x86_64
Jul-17 01:15:48.358 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onTorNodeReady 
Jul-17 01:15:48.359 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: initWalletService 
Jul-17 01:15:48.395 [ STARTING] INFO  i.b.c.b.w.WalletConfig: Wallet directory: /home/user/.local/share/bisq/btc_mainnet/wallet 
Jul-17 01:15:49.048 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: walletsSetup.onInitialized 
Jul-17 01:15:54.661 [JavaFX Application Thread] INFO  i.b.n.p.p.g.RequestDataHandler: 
#################################################################
Received 67 instances of storage payload
MailboxStoragePayload: 6
Alert: 1
Arbitrator: 1
OfferPayload: 59
################################################################# 
Jul-17 01:15:54.887 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onRequestingDataCompleted 
Jul-17 01:15:54.888 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onBasicServicesInitialized 
Jul-17 01:15:55.083 [Save-file-task-4548] INFO  i.b.c.s.FileManager: Write to disc: UserPayload 
Jul-17 01:15:56.694 [Save-file-task-2206] INFO  i.b.c.s.FileManager: Write to disc: SequenceNumberMap 
Jul-17 01:15:57.365 [Save-file-task-6746] INFO  i.b.c.s.FileManager: Write to disc: NavigationPath 
Jul-17 01:15:57.748 [JavaFX Application Thread] INFO  i.b.c.p.f.FeeService: BTC tx fee: txFeePerByte=80 
Jul-17 01:15:58.125 [Save-file-task-3958] INFO  i.b.c.s.FileManager: Write to disc: PreferencesPayload 
Jul-17 01:15:58.342 [Save-file-task-5767] INFO  i.b.c.s.FileManager: Write to disc: TradeStatistics 
Jul-17 01:16:21.779 [Thread-9] INFO  c.m.t.t.OnionProxyManagerEventHandler: Hidden service xxx.onion:9999 published. 
Jul-17 01:16:21.805 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onHiddenServicePublished 
Jul-17 01:16:24.130 [JavaFX Application Thread] INFO  i.b.n.p.p.g.RequestDataHandler: 
#################################################################
Received 66 instances of storage payload
MailboxStoragePayload: 6
Alert: 1
Arbitrator: 1
OfferPayload: 58
################################################################# 
Jul-17 01:16:24.313 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onBootstrapComplete 
Jul-17 01:16:24.321 [JavaFX Application Thread] INFO  i.b.g.m.MainViewModel: onRequestingDataCompleted 
Jul-17 01:16:26.182 [Save-file-task-9958] INFO  i.b.c.s.FileManager: Write to disc: SequenceNumberMap 
Jul-17 01:16:27.662 [Save-file-task-129] INFO  i.b.c.s.FileManager: Write to disc: PeerList 
Jul-17 01:16:31.934 [Save-file-task-1124] INFO  i.b.c.s.FileManager: Write to disc: PeerList 
Jul-17 01:16:34.442 [Save-file-task-121] INFO  i.b.c.s.FileManager: Write to disc: SequenceNumberMap 
Jul-17 01:16:37.005 [Save-file-task-872] INFO  i.b.c.s.FileManager: Write to disc: PeerList 
Jul-17 01:16:41.242 [Save-file-task-8046] INFO  i.b.c.s.FileManager: Write to disc: PeerList 
Jul-17 01:16:41.735 [Save-file-task-5894] INFO  i.b.c.s.FileManager: Write to disc: SequenceNumberMap 
Jul-17 01:16:48.894 [Save-file-task-2863] INFO  i.b.c.s.FileManager: Write to disc: PreferencesPayload 
Jul-17 01:16:50.601 [Save-file-task-960] INFO  i.b.c.s.FileManager: Write to disc: SequenceNumberMap 
...
```

**Comment by ManfredKarrer** [comment]
Thats very strange.... Can you check if the app data dir (.local/share/bisq) is the same as before?
can u send me the full log file by email (manfred at bitsquare dot io)?

**Comment by nothingmuch** [comment]
It appears you've already identified the underlying problem and fixed it with 0.5.3 since commenting, I can confirm it was the ~/.local/share/{b,B}isq issue (didn't spot that in the logs myself!), and that 0.5.3 loads the 0.5.1 wallet
